src/transactions/test/ChangeTrustTests.cpp

```diff
@@ -268,6 +268,136 @@ TEST_CASE("change trust", "[tx][changetrust]")
+                REQUIRE_THROWS_AS(root.changeTrust(poolShareAsset, 10),
+                                  ex_CHANGE_TRUST_TRUST_LINE_MISSING);
+                root.changeTrust(assetB, 10);
+
+                // assetB trustline is not authorized to maintain liabilities
+                REQUIRE_THROWS_AS(
+                    root.changeTrust(poolShareAsset, 10),
+                    ex_CHANGE_TRUST_NOT_AUTH_MAINTAIN_LIABILITIES);
+                gateway.allowMaintainLiabilities(assetB, root);
+            }
+
+            // this should create a LiquidityPoolEntry, and modify
+            // liquidityPoolUseCount on the asset trustlines
+            auto prePoolNumSubEntries = getNumSubEntries(root);
+            root.changeTrust(poolShareAsset, 10);
+
+            // TODO: This line requires the update to SponsorshipUtils
+            // REQUIRE(getNumSubEntries(root) - prePoolNumSubEntries == 2);
+
+            auto poolShareTlAsset =
+                changeTrustAssetToTrustLineAsset(poolShareAsset);
+
+            // pool share trustline shouldn't have any flags set
+            REQUIRE(root.loadTrustLine(poolShareTlAsset).flags == 0);
+
+            if (assetA.type() != ASSET_TYPE_NATIVE)
+            {
+                auto assetATl = root.loadTrustLine(assetA);
+                REQUIRE(getTrustLineEntryExtensionV2(assetATl)
+                            .liquidityPoolUseCount == 1);
+            }
+
+            if (assetB.type() != ASSET_TYPE_NATIVE)
+            {
+                auto assetBTl = root.loadTrustLine(assetB);
+                REQUIRE(getTrustLineEntryExtensionV2(assetBTl)
+                            .liquidityPoolUseCount == 1);
+            }
+
+            {
+                LedgerTxn ltx(app->getLedgerTxnRoot());
+                auto pool =
+                    loadLiquidityPool(ltx, poolShareTlAsset.liquidityPoolID());
+                REQUIRE(pool);
+                REQUIRE(pool.current()
+                            .data.liquidityPool()
+                            .body.constantProduct()
+                            .poolSharesTrustLineCount == 1);
+            }
+
+            // can't delete asset trustlines while they are used in a pool
+            if (assetA.type() != ASSET_TYPE_NATIVE)
+            {
+                REQUIRE_THROWS_AS(root.changeTrust(assetA, 0),
+                                  ex_CHANGE_TRUST_CANNOT_DELETE);
+            }
+            if (assetB.type() != ASSET_TYPE_NATIVE)
+            {
+                REQUIRE_THROWS_AS(root.changeTrust(assetB, 0),
+                                  ex_CHANGE_TRUST_CANNOT_DELETE);
+            }
+
+            // delete the pool sharetrust line
+            root.changeTrust(poolShareAsset, 0);
+            REQUIRE(!root.hasTrustLine(poolShareTlAsset));
+
+            // TODO: This line requires the update to SponsorshipUtils
+            // REQUIRE(getNumSubEntries(root) == prePoolNumSubEntries);
```

Comment:
Same as above

Reply:
Normally I'm opposed to TODO's, but I recognize why they are here. Open an issue for them once this gets merged since that's the easier way to remember to do things.